Source/Urho3D/IO/File.cpp

```diff
@@ -136,21 +301,44 @@ bool File::Open(PackageFile* package, const ea::string& fileName)
     if (!entry)
         return false;
 
-    bool success = OpenInternal(package->GetName(), FILE_READ, true);
+    bool success = OpenInternal(package->GetName(), FILE_READ, entry, package->PackageEncoding() != PE_NONE);
     if (!success)
     {
         URHO3D_LOGERROR("Could not open package file " + fileName);
         return false;
     }
 
     name_ = fileName;
-    offset_ = entry->offset_;
     checksum_ = entry->checksum_;
     size_ = entry->size_;
-    compressed_ = package->IsCompressed();
 
-    // Seek to beginning of package entry's file data
-    SeekInternal(offset_);
+    switch (package->PackageEncoding())
+    {
+    case PE_NONE:
+        break;
+    case PE_LZ4:
+        decoder_ = ea::make_unique<CompressedStreamDeserializer>(file_);
+        break;
+    case PE_TWEETNACL:
+        if (!key)
+        {
+            URHO3D_LOGERROR("No encryption key provided");
+            return false;
+        }
+        decoder_ = ea::make_unique<EncryptedStreamDeserializer>(file_, *key);
+        break;
+    case PE_LZ4_TWEETNACL:
+        if (!key)
+        {
+            URHO3D_LOGERROR("No encryption key provided");
+            return false;
+        }
+        decoder_ = ea::make_unique<EncryptedStreamDeserializer>(file_, *key);
+        break;
```

Comment:
Seems like in this case `PackageEncoding` could be flags. `PE_COMPRESSED` and `PE_ENCRYPTED`. There is no need to reflect underlying algorithms in enum names as i doubt there is any need to introduce any alternative algorithms. However should anyone find it necessary - `PE_COMPRESSED` and `PE_ENCRYPTED` would serve as good names for defaults, whole additional algorithms could include their names in extra enum values.

Reply:
Is it not flags now? I've tried to replicate common Urho pattern for flags here.